docker/docker_client.go

```diff
@@ -103,11 +103,47 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 		tlsc := &tls.Config{}
 
 		if ctx.DockerCertPath != "" {
-			cert, err := tls.LoadX509KeyPair(filepath.Join(ctx.DockerCertPath, "cert.pem"), filepath.Join(ctx.DockerCertPath, "key.pem"))
-			if err != nil {
-				return nil, fmt.Errorf("Error loading x509 key pair: %s", err)
+			fs, err := ioutil.ReadDir(ctx.DockerCertPath)
+			if err != nil && !os.IsNotExist(err) {
+				return nil, err
+			}
+
+			for _, f := range fs {
```

Comment:
(Adding a `fullPath := filepath.Join(ctx.DockerCertPath, f.Name())` here would avoid some of the repetition, but this is perfectly fine as well.)